tag-historian/nitaghistorian.yml

```diff
@@ -0,0 +1,928 @@
+          name: path
+          type: string
+          required: true
+          description: The path of the tag that owns the historical value being deleted.
+          x-example: system1.tag1
+        - in: path
+          name: id
+          type: string
+          required: true
+          description: The ID of the historical value to delete.
+          x-example: '5c88d2ae26b41e22f08b4407'
+      responses:
+        204:
+          description: Success
+        401:
+          $ref: '#/responses/Unauthorized'
+        default:
+          $ref: '#/responses/Error'
+  /v1/tags/delete-values-by-range:
+    post:
+      tags: [history]
+      summary: Deletes a range of historical values
+      description: Deletes all historical values within a given time range for one or more tags.
+      operationId: DeleteHistoricalValuesByRange
+      x-ni-operation: deleteHistoricalTagValues
+      parameters:
+        - in: body
+          name: Request body
+          required: true
+          description: Contains the time ranges to delete.
+          schema:
+            description: Contains the time ranges to delete.
+            type: object
+            title: Tag Historian Delete Values By Range Request
+            properties:
+              ranges:
+                description: The time ranges to delete.
+                type: array
+                items:
+                  description: A time range of historical tag values to delete.
+                  type: object
+                  title: Historical Tag Query Range
+                  properties:
+                    path:
+                      description: The path of the tag whose historical values within the range will be deleted.
+                      type: string
+                      example: system1.tag1
+                    startTime:
+                      description: The ISO-8601 formatted date and time indicating the start of the range. Any historical values from this time until endTime will be deleted.
+                      type: string
+                      format: iso-date-time
+                      example: '2019-03-13T09:51:38Z'
+                    endTime:
+                      description: The ISO-8601 formatted date and time indicating the end of the range. Any historical values from startTime to and including this time will be deleted.
+                      type: string
+                      format: iso-date-time
+                      example: '2020-03-13T09:51:38Z'
+            required: [ranges]
+      responses:
+        204:
+          description: Success
+        401:
+          $ref: '#/responses/Unauthorized'
+        default:
+          $ref: '#/responses/Error'
+  /v1/tags/flush-values:
+    post:
+      tags: [utility]
+      summary: Process buffered tag updates
+      description: Triggers the tag historian to immediately begin processing tag updates. Clients do not need to call this method for updates to be processed normally.
```

Comment:
Do you think it would be valuable to call out the default batch interval? It's probably something like every 5 minutes or 1000 updates, whichever comes first.

Reply:
Well, the route is really only meant for testing and not something we expect clients to ever call. Updates are something like 5 seconds, and aging is like an hour. I don't really want to try to pin down the exact numbers here.

Reply:
Sounds reasonable.